src/lib.rs

```diff
@@ -412,6 +420,50 @@ fn run_for_context<TContext: Sync>(
     })
 }
 
+pub fn run_for_slice_with_callback(
+    slice: &[u8],
+    args: Args,
+    mut callback: impl FnMut(CaptureInfo) + Sync,
```

Comment:
(Vs the things that get passed to the callback in the `run_with_callback()` entry point) having the callback get passed the slice and a path didn't seem to make sense for this use case so it just gets passed the "capture info"